metricbeat/module/mongodb/mongodb.go

```diff
@@ -155,33 +98,51 @@ func NewClient(config ModuleConfig, timeout time.Duration, mode readpref.Mode) (
 		}
 	}
 
-	if mode == 0 {
-		mode = readpref.NearestMode
+	if config.TLS.IsEnabled() {
+		tlsConfig, err := tlscommon.LoadTLSConfig(config.TLS)
+		if err != nil {
+			return nil, fmt.Errorf("could not load provided TLS configuration: %w", err)
+		}
+
+		clientOptions.SetTLSConfig(tlsConfig.ToConfig())
+	}
+
+	host := base.Host()
+	if parts := strings.SplitN(host, "://", 2); len(parts) != 2 {
+		host = "mongodb://" + host // add scheme
+	}
+	if err := clientOptions.ApplyURI(host).Validate(); err != nil {
```

Comment:
`ApplyURI` is purposedly put here because as per the documentation, any methods on `clientOptions` i.e, of `SetXXX` form overrides the URI parameters (if any) and vice-versa. So it would give the user freedom to set as desired parameters using the URI itself and it overrides the values of the fields that were set using `SetXXX`